src/value/mod.rs

```diff
@@ -335,7 +348,14 @@ pub trait Struct {
     /// [gat]: https://rust-lang.github.io/rfcs/1598-generic_associated_types.html
     fn iter<'a>(
         &'a self,
-    ) -> Box<dyn Iterator<Item = (&'a Self::FieldName, &'a Self::Element)> + 'a>;
+    ) -> Box<dyn Iterator<Item = (&'a Self::FieldName, Option<&'a Self::Element>)> + 'a>;
+
```

Comment:
Sorry. This seems incorrect I guess. I would assume that `fields`  would be a `HashMap` with `(field_name, values)` where `values` can not be empty(returning `None` here) as a field would only be added if it has at least one value. does that answer your question?

Reply:
I might be not asking the question well. Let me rephrase, There seems to be a change here to the iterator in which the items are pairs of `&Self::FieldName` and `Option<&Element>`.  Why `Option<&Self::Element>` or more specifically what does it mean to iterate over a struct and get some field name with `None` instead of `Some<&Self::Element>`?